code/datums/storage/storage.dm

```diff
@@ -161,7 +161,11 @@
 /datum/storage/proc/on_deconstruct()
 	SIGNAL_HANDLER
 
-	remove_all()
+	var/atom/drop_location = real_location.drop_location()
+	for(var/obj/thing in real_location)
+		thing.forceMove(drop_location)
+		thing.pixel_x = thing.base_pixel_x + rand(-8, 8)
+		thing.pixel_y = thing.base_pixel_y + rand(-8, 8)
```

Comment:
You should make remove_all() pass silent and ignore animations and views instead of cutting it out entirely, as ``attempt_remove`` handles internal mob storages (and dropped() calls in that scenario) and item resets, you're risking messing up planes and maptext here.

Reply:
Thats a good point